daphne_worker_test/tests/e2e.rs

```diff
@@ -537,3 +540,94 @@ async fn e2e_leader_collect_abort_invalid_batch_interval() {
     )
     .await;
 }
+
+#[tokio::test]
+#[cfg_attr(not(feature = "test_e2e"), ignore)]
+async fn e2e_leader_collect_abort_overlapping_batch_interval() {
+    let t = TestRunner::default().await;
+    let batch_interval = t.batch_interval();
+
+    let client = t.http_client();
+    let hpke_config_list = t.get_hpke_configs(&client).await;
+
+    // The reports are uploaded in the background.
+    let mut rng = thread_rng();
+    for _ in 0..t.min_batch_size {
+        let now = rng.gen_range(batch_interval.start..batch_interval.end());
+        t.leader_post_expect_ok(
+            &client,
+            "/upload",
+            constants::MEDIA_TYPE_REPORT,
+            t.vdaf
+                .produce_report(&hpke_config_list, now, &t.task_id, DapMeasurement::U64(1))
+                .unwrap()
+                .get_encoded(),
+        )
+        .await;
+    }
+
+    // Get the collect URI.
+    let collect_req = CollectReq {
+        task_id: t.task_id.clone(),
+        batch_interval: batch_interval.clone(),
+        agg_param: Vec::new(),
+    };
+    let _collect_uri = t
+        .leader_post_collect(&client, collect_req.get_encoded())
+        .await;
+
+    // The reports are aggregated in the background.
+    let agg_telem = t
+        .internal_process(
+            &client,
+            &InternalAggregateInfo {
+                max_buckets: 100, // Needs to be sufficiently large to touch each bucket.
+                max_reports: 100,
+            },
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        agg_telem.reports_processed, t.min_batch_size,
+        "reports processed"
+    );
+    assert_eq!(
+        agg_telem.reports_aggregated, t.min_batch_size,
+        "reports aggregated"
+    );
+    assert_eq!(
+        agg_telem.reports_collected, t.min_batch_size,
+        "reports collected"
+    );
+
+    // Send a collect request that requests results from batches that overlaps with
+    // the previous request.
+    //
+    // NOTE: Since DURABLE_LEADER_COL_JOB_QUEUE_PUT has a mechanism to reject CollectReq
+    // with the EXACT SAME content as previous requests, we need to tweak the request
+    // a little bit.
+    let collect_req = CollectReq {
+        task_id: t.task_id.clone(),
+        batch_interval: Interval {
+            start: batch_interval.start,
+            duration: batch_interval.duration * 2,
+        },
+        agg_param: Vec::new(),
+    };
+    let _collect_uri = t
+        .leader_post_collect(&client, collect_req.get_encoded())
+        .await;
+
+    // The reports are aggregated in the background.
+    let err = t
+        .internal_process(
+            &client,
+            &InternalAggregateInfo {
+                max_buckets: 100, // Needs to be sufficiently large to touch each bucket.
+                max_reports: 100,
+            },
+        )
+        .await
+        .unwrap_err();
+    assert_matches!(err, DapAbort::BatchOverlap);
```

Comment:
Hmm, this is odd. I would expect `POST /internal/process` to succeed (although no reports would be aggregated). Conversely, I would expect the second `POST /collect`above to fail. In fact, this is what the spec mandates.

Reply:
I've updated the code so that the collect req gets checked when it gets posted.